File: 30daysOFcode_Java/day 17.java
import java.util.*;
import java.io.*;
import java.math.*;
//Write your code here

class Calculator extends Exception
{
    int power (int n,int p) throws Exception {
      
        if(n<0 || p<0)
           { String ans ="n and p should be non-negative"; throw new Exception(ans) ;}
            


       else return (int)Math.pow(n, p);

     // return 1;
    }
}

class Solution{

File: 30daysOFcode_C++/day 17.cpp
#include <cmath>
#include <iostream>
#include <exception>
#include <stdexcept>
using namespace std;

//Write your code here
class Calculator {
public:
  int power(int n, int p)
  {
    if (n < 0 || p < 0)
    {
      throw runtime_error("n and p should be non-negative");
    }
    return pow(n, p);
  }
};

int main()
{
  Calculator myCalculator = Calculator();
  int T, n, p;
  cin >> T;
  while (T-- > 0) {
    if (scanf("%d %d", &n, &p) == 2) {
      try {
        int ans = myCalculator.power(n, p);
        cout << ans << endl;
      }
      catch (exception& e) {
        cout << e.what() << endl;
      }
    }
  }

}